src/painter/config.rs

```diff
@@ -169,12 +170,34 @@ impl FromWorld for ShapeConfig {
     }
 }
 
+pub struct ShapeConfigState {
+    shape_config: SyncCell<ShapeConfig>,
+    resource_state: <Res<'static, BaseShapeConfig> as SystemParam>::State,
+}
+
 unsafe impl SystemParam for &mut ShapeConfig {
-    type State = SyncCell<ShapeConfig>;
+    type State = ShapeConfigState;
     type Item<'w, 's> = &'s mut ShapeConfig;
 
-    fn init_state(world: &mut World, _system_meta: &mut SystemMeta) -> Self::State {
-        SyncCell::new(ShapeConfig::from_world(world))
+    fn init_state(world: &mut World) -> Self::State {
+        ShapeConfigState {
+            shape_config: SyncCell::new(ShapeConfig::from_world(world)),
+            resource_state: Res::<BaseShapeConfig>::init_state(world),
+        }
+    }
+
+    fn init_access(
+        state: &Self::State,
+        system_meta: &mut SystemMeta,
+        component_access_set: &mut FilteredAccessSet,
+        world: &mut World,
+    ) {
+        Res::<BaseShapeConfig>::init_access(
+            &state.resource_state,
+            system_meta,
+            component_access_set,
+            world,
+        );
```

Comment:
Based on https://bevy.org/learn/migration-guides/0-16-to-0-17/#stop-storing-access-in-systems

I think this is right, but I'm not an expert.